app/scripts/lib/extension-port-stream.ts

```diff
@@ -0,0 +1,398 @@
+/* eslint-disable no-plusplus, no-bitwise, default-case */
+
+import type { Json } from '@metamask/utils';
+import { Duplex, type DuplexOptions } from 'readable-stream';
+import type { Runtime } from 'webextension-polyfill';
+
+let _randomCounter = 0 | 0;
+/**
+ * Creates a random ID for chunked messages. Always <= Int32 max value.
+ *
+ * @returns a random number that can be used as an ID
+ */
+function getNextId(): number {
+  // increment the counter and wrap around if it exceeds Int32 max value
+  _randomCounter = (_randomCounter + 1) & 0x7fffffff;
```

Comment:
So, we have to clamp the value to _something_, and Number.MAX_SAFE_INTEGER is a bit overkill, so why not clamp it to something that also gives the runtime compiler an optimization hint: int32!

We later use the fact that IDs can't exceed max int32 when parsing chunk frames to use a much more optimal number parsing mechanism (than parseInt and character checks).